docs/tutorials/comet-impact.ipynb

```diff
@@ -0,0 +1,410 @@
+    "We can create a 1-D variable of velocities:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "de078215-f87d-433e-ba7c-94be6254c77d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "v = sc.array(dims=['velocity'], values=[20., 30., 40., 50.], unit='km/s')\n",
+    "v"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5b20ed91-adf0-47c4-9f8e-0970b5263f44",
+   "metadata": {},
+   "source": [
+    "In contrast to a plain NumPy array, the scipp variable also stores a unit and a tuple of dimension labels.\n",
+    "We can use this array of velocities to compute multiple temperature increases at a time:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "72c8acd8-0646-45e2-994c-123590c5c22a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "temperature_increase(rho=rho, r=r, v=v)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f9b92820-29df-44f2-ad8e-cdb00445006b",
+   "metadata": {},
+   "source": [
+    "We could also define multiple comet radii and compute:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7c68e2cf-8946-4f39-a3f0-31e7f862ade5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "r = sc.array(dims=['radius'], values=[2.0, 4.0, 8.0], unit='km')\n",
+    "temp = temperature_increase(rho=rho, r=r, v=v)\n",
+    "temp"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dd455367-ee04-4306-b4e8-b712c3959e68",
+   "metadata": {},
+   "source": [
+    "The function now returns a 2-D variable.\n",
+    "Since `r` and `v` have different dimension labels scipp automatically computes a value for each possible combination.\n",
+    "This is called *broadcasting*.\n",
+    "\n",
+    "The dimension labels can be used, e.g., to specify which dimension to slice.\n",
+    "We can obtain a view for the second radius value using:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9430b3ec-fe1b-4fa9-98f7-1c30e306dc1d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "temp['radius', 1]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "89fca0e6-35eb-4aaa-a82c-1b07d6647d6a",
+   "metadata": {},
+   "source": [
+    "One thing is missing from our result:\n",
+    "It is difficult to identify which data value belong to which value of `r` and `v`.\n",
+    "This can be achieved by storing them as *coordinates* of the data, using scipp's `DataArray`, which is covered in other tutorials."
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3 (ipykernel)",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 5
+}
```

Comment:
I'm not sure how much this teaches you about Scipp. It's a nice problem to work out with the units, but basically most of it is just showing you that you can operate with quantities that have units (apart from the end where you show that you can also use arrays). One could almost do the same tutorial with `pint`.

I agree that the current 'getting started' tutorial is very old, the exercises could be better, and it seems a bit backwards to start with a dataset.

But I am thinking that this proposed one is not giving you the tools to understand the other tutorials and docs (coordinates, data arrays, bin edges, etc.).
Currently you would need to start with this one, then read the https://scipp.github.io/user-guide/data-structures.html page, before moving on to other tutorials/docs.

Maybe if we actually introduced data arrays at the end, and then e.g. make a 2d plot of the result, it would address most of my concerns? Then you could further illustrate the slicing you introduced with 1d plots that show the effect of the different parameters? I think coordinates and slicing are some of the most important concepts to talk about.

Reply:
Maybe bottom-up is indeed the wrong approach (for a *tutorial*)? The user-guide and reference docs can be seen as bottom-up, maybe tutorials should to top-down? This is, should we load some 2-D (or 3-D) dense data and go from there?

Reply:
Personally, I prefer a bottom-up approach. But it needs to motivate and explain adding more layers. So like Neil said, with the current tutorials, this is missing. We have the bottom layers and the top layers but they are not really connected.

Reply:
That is, bottom-up like in the previous (closed) PR until we have a data array, but more motivated on an actual example?

Reply:
Pretty much